src/main/java/matching_algorithm_use_case/MatchingAlgorithmOutputBoundary.java

```diff
@@ -5,7 +5,7 @@
 public interface MatchingAlgorithmOutputBoundary {
 
     /**
-     * Send the response model (success message and ordered lists) to the screens
+     * Send the response model (success message and ordered list) to the screens
      * @param responseModel: Update message and ordered List of Groups as Strings
      * @return an instance of the Matching Algorithm response model
      */
```

Comment:
Does this run without error? The prepareFailView method is returning a MatchingAlgorithmResponseModel, which is not a string to my knowledge.